app/game-engine/src/main/kotlin/opensamguk/engine/v2/V2ProcessCityIncome.kt

```
@@ -0,0 +1,195 @@
+    val generalPayouts: List<IncomeGeneralPayout>,
+    val globalHistory: String,
+)
+
+/**
+ * 도시 단위 수입 정산. 국가 id 오름차순 → 도시 id 오름차순(결정론 고정, v1의 `SELECT` 행 순서 관례와 동형).
+ *
+ * `prevIncome`은 **국가 단위로 유지**한다(설계 §2.3 판정 (a)) — 값은 그 국가 도시들의 수입 합.
+ * `nation_env.prev_income_{gold,rice}` KV 소비처(랭킹·내정 화면)가 국가 단위라서 도시별로 쪼개면
+ * 소비처가 전부 깨진다.
+ */
+fun v2ProcessCityIncome(
```

detekt (1.23.8)

[warning] The function v2ProcessCityIncome appears to be too complex based on Cognitive Complexity (complexity: 49). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)